packages/front-end/pages/oauth/callback.tsx

```diff
@@ -25,7 +25,9 @@ export default function OAuthCallbackPage() {
         if (json?.status !== 200) {
           setError(json?.message || "An unknown error occurred");
         } else {
-          router.replace("/");
+          const redirect =
+            window.sessionStorage.getItem("postAuthRedirectPath") ?? "/";
+          router.replace(redirect);
```

Comment:
Need to:
1. Wrap in a try/catch (sessionStorage throws an error sometimes, e.g. in incognito windows)
2. Verify it's a relative path before redirecting (starts with a slash and then a letter)